docs/csharp/language-reference/operators/and-operator.md

```diff
@@ -11,24 +11,52 @@ helpviewer_keywords:
-  
-## Remarks  
- The unary `&` operator returns the address of its operand (requires [unsafe](../../../csharp/language-reference/keywords/unsafe.md) context).  
-  
- Binary `&` operators are predefined for the integral types and `bool`. For integral types, & computes the logical bitwise AND of its operands. For `bool` operands, & computes the logical AND of its operands; that is, the result is `true` if and only if both its operands are `true`.  
-  
- The binary `&` operator evaluates both operands regardless of the first one's value, in contrast to the [conditional AND operator](../../../csharp/language-reference/operators/conditional-and-operator.md) `&&`. For example:  
-  
- [!code-csharp[csRefOperators#37](../../../csharp/language-reference/operators/codesnippet/CSharp/and-operator_1.cs)]  
-  
- User-defined types can overload the binary `&` operator (see [operator](../../../csharp/language-reference/keywords/operator.md)). Operations on integral types are generally allowed on enumeration. When a binary operator is overloaded, the corresponding assignment operator, if any, is also implicitly overloaded.  
-  
-## Example  
- [!code-csharp[csRefOperators#38](../../../csharp/language-reference/operators/codesnippet/CSharp/and-operator_2.cs)]  
-  
-## See Also
-
-- [C# Reference](../../../csharp/language-reference/index.md)  
-- [C# Programming Guide](../../../csharp/programming-guide/index.md)  
-- [C# Operators](../../../csharp/language-reference/operators/index.md)
+
+The `&` operator is supported in two forms: a unary address-of operator or a binary logical operator.
+
+## Unary address-of operator
+
+The unary `&` operator returns the address of its operand. For more information, see [How to: obtain the address of a variable](../../programming-guide/unsafe-code-pointers/how-to-obtain-the-address-of-a-variable.md).
+
+The address-of operator `&` requires [unsafe](../keywords/unsafe.md) context.
+
+## Integer logical bitwise AND operator
+
+For integer types, the `&` operator computes the logical bitwise AND of its operands:
+
+[!code-csharp-interactive[integer logical bitwise AND](~/samples/snippets/csharp/language-reference/operators/AndOperatorExamples.cs#IntegerOperands)]
+
+> [!NOTE]
+> The preceding example uses the binary literals that are [introduced](../../whats-new/csharp-7.md#numeric-literal-syntax-improvements) in C# 7.0 and [enhanced](../../whats-new/csharp-7-2.md#leading-underscores-in-numeric-literals) in C# 7.2.
+
+Because operations on integer types are generally allowed on enumeration types, the `&` operator also supports [enum](../keywords/enum.md) operands.
+
+## Boolean logical AND operator
+
+For [bool](../keywords/bool.md) operands, the `&` operator computes the logical AND of its operands. The result of `x & y` is `true` if both `x` and `y` are `true`. Otherwise, the result is `false`.
+
+The `&` operator evaluates both operands regardless of the value of its first operand, as the following example shows:
```

Comment:
I'm wondering whether this could use some additional detail or rephrasing. For example, "The & operator evaluates both operands even if the value of the first operand is false, so that the result must be false." My concern here is that inexperienced developers may not realize what this means.